### PR TITLE
chore: cleanup and upgrade some api to v1

### LINKED ISSUE #3625: Clean-up `gwapiv1a2.ParentRef`
The updates in `GetRouteParentContext` and `GetParentReferences` is indeed necessary, since  the `ParentRef` for xRoute got all upgraded to `v1`.

So the codes here do need a clean-up, but we can track this in a separate issue.

_Originally posted by @shawnh2 in https://github.com/envoyproxy/gateway/pull/3532#discussion_r1645356783_

---

clean-up the code under `gatewayapi`, upgrade `gwapiv1a1.ParentRef` to `gwapiv1.ParentRef`.

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3625

### CODE REVIEW

File: internal/gatewayapi/contexts.go [hunk 1]
```diff
@@ -275,27 +273,27 @@ func GetRouteParentContext(route RouteContext, forParentRef gwapiv1.ParentRefere
...
 		if reflect.DeepEqual(p, forParentRef) {
 			routeParentStatusIdx = i
 			break
 		}
 	}
+
 	if routeParentStatusIdx == -1 {
 		tmpPR := forParentRef
-		if !isHTTPRoute {
+		if !isV1Route {
```
Comment (arkodg) [inline internal/gatewayapi/contexts.go:296]: do we need this code now that ParentReference is v1 ?

Comment (shawnh2) [reply internal/gatewayapi/contexts.go]: no need, removed

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]

**Pull request merged**